test/AutoDiff/SILGen/autodiff_builtins.swift

```diff
@@ -84,24 +84,19 @@ func applyDerivative_f1_vjp<T: AdditiveArithmetic & Differentiable>(t0: T) -> (T
 // CHECK: copy_addr [take] [[D_RESULT_BUFFER_0_FOR_LOAD]] to [init] [[ORIG_RESULT_OUT_PARAM]]
 // CHECK: return [[PULLBACK]]
 
-struct ExamplePullbackStruct<T: Differentiable> {
-  var pb0: (T.TangentVector) -> T.TangentVector
-}
-
 @_silgen_name("test_context_builtins")
-func test_context_builtins() {
-  let pbStruct = ExamplePullbackStruct<Float>(pb0: { $0 })
-  let context = Builtin.autoDiffCreateLinearMapContext(Builtin.sizeof(type(of: pbStruct)))
+func test_context_builtins<T>(t: T) {
+  let context = Builtin.autoDiffCreateLinearMapContext(T.self)
   let topLevelSubctxAddr = Builtin.autoDiffProjectTopLevelSubcontext(context)
-  UnsafeMutableRawPointer(topLevelSubctxAddr).storeBytes(of: pbStruct, as: type(of: pbStruct))
-  let newBuffer = Builtin.autoDiffAllocateSubcontext(context, Builtin.sizeof(type(of: pbStruct)))
-  UnsafeMutableRawPointer(newBuffer).storeBytes(of: pbStruct, as: type(of: pbStruct))
+  UnsafeMutableRawPointer(topLevelSubctxAddr).storeBytes(of: t, as: T.self)
+  let newBuffer = Builtin.autoDiffAllocateSubcontext(context, T.self)
+  UnsafeMutableRawPointer(newBuffer).storeBytes(of: t, as: T.self)
 }
 
-// CHECK-LABEL: sil{{.*}}@test_context_builtins
-// CHECK: bb0:
-// CHECK:   [[CTX:%.*]] = builtin "autoDiffCreateLinearMapContext"({{%.*}} : $Builtin.Word) : $Builtin.NativeObject
-// CHECK:   [[BORROWED_CTX:%.*]] = begin_borrow [lexical] [[CTX]] : $Builtin.NativeObject
-// CHECK:   [[BUF:%.*]] = builtin "autoDiffProjectTopLevelSubcontext"([[BORROWED_CTX]] : $Builtin.NativeObject) : $Builtin.RawPointer
-// CHECK:   [[BUF:%.*]] = builtin "autoDiffAllocateSubcontext"([[BORROWED_CTX]] : $Builtin.NativeObject, {{.*}} : $Builtin.Word) : $Builtin.RawPointer
+// CHECK-LABEL: sil{{.*}}@test_context_builtins : $@convention(thin) <T> (@in_guaranteed T) -> () {
+// CHECK: bb0({{%.*}} : $*T):
+// CHECK:   [[CTX:%.*]] = builtin "autoDiffCreateLinearMapContext"<T>({{%.*}} : $@thick T.Type) : $Builtin.NativeObject // users: {{.*}}
```

Comment:
Is it really necessary to make these builtins generic?

Reply:
So the actual builtins are actually not generic. I'm not entirely sure why, but I simply changed the `size_t` arg to a metatype and this is the code that the compiler is generating. Even the SIL that's generated from source does not have generic builtins.

Reply:
On a side note, I had a hard time figuring out how to get a thick metatype for a non-class type, which is why I had to make the test function generic. Would you know if it's possible to get a thick metatype for a struct/enum here?

Reply:
Ah, sorry, I misread the test changes. It's the test function `test_context_builtins` that is now generic!